### PR TITLE
GD-9068: Security improvements

### PR DESCRIPTION
**Pull request by mihailradkov**
GDB-9068: Update the admin user default details

- Added `security.admin` for configuring the username and/or the password of the administrator user
- Added `security.users` for inserting additional users into the default initial user.js configuration

GDB-9440: Add resource limits for all containers

- Added default resource limits and requests for all init containers and provisioning jobs

GDB-9437: Enabled the Security Context by default

- Enabled security context by default for all pods and containers
- Added `initContainerDataPermissions` and `proxy.initContainerDataPermissions` for changing permissions in the storage volumes if needed
- Updated the chart to require Kubernetes version 1.26+

Added configurations for extra PVC templates

- Added `extraVolumeClaimTemplates` and `proxy.extraVolumeClaimTemplates`

Added support for emptyDir persistence

- Moved `persistence.volumeClaimTemplateSpec` to `persistence.volumeClaimTemplate.spec`
- Added `persistence.emptyDir` and `proxy.persistence.emptyDir` configurations for an emptyDir volume that will be used when the persistence is disabled
- Added configurations for extra `labels` and `annotations` for all persistent volume claim templates: `persistence.volumeClaimTemplate`, `proxy.persistence.volumeClaimTemplate`  and `import.volumeMount.volumeClaimTemplate`

Additionally

- Moved initContainers declaration to be before the actual containers in the pod
- Set image digest by default
- Enabled PodDisruptionBudget by default

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -357,13 +380,40 @@ podAnnotations: {}
...
+    type: RuntimeDefault
 
 # Defines privilege and access control settings for the node container running GraphDB.
-securityContext: {}
+securityContext:
+  allowPrivilegeEscalation: false
+  readOnlyRootFilesystem: true
+  capabilities:
+    drop: [ "ALL" ]
+  seccompProfile:
```
Comment (mihailradkov) [inline values.yaml:398]: I put `seccompProfile` in both the pod and the container contexts cause some tools were complaining. TBH, I think that having the profile in `podSecurityContext` is enough.

File: values.yaml [hunk 2]
```diff
@@ -387,12 +437,22 @@ topologySpreadConstraints: []
 resources:
   limits:
     memory: 2Gi
+    ephemeral-storage: 1Mi
```
Comment (mihailradkov) [inline values.yaml:440]: This came from kube-score, it complains that both limits and requests should have some value for the ephemeral storage... WDYT @yaskoo

**Review by yaskoo — APPROVED** [review]

File: values.yaml (hunk 2 above)
Comment (mihailradkov) [reply values.yaml]: Decided to remove them for now.

File: templates/proxy/statefulset.yaml [hunk 3]
```diff
@@ -207,7 +180,7 @@ spec:
               mountPath: /tmp/graphdb/graphdb-extra-secret.properties
               subPath: {{ .Values.proxy.extraConfiguration.properties.secretKey }}
             {{- end }}
-          command: [ 'sh', '-c' ]
+          command: [ "sh", "-c" ]
```
[the remark this answers is not in the document]
Comment (mihailradkov) [reply templates/proxy/statefulset.yaml]: Changed

File: values.yaml (hunk 1 above)
Comment (mihailradkov) [reply values.yaml:397]: Lets leave them both for now.

**Pull request merged**